substrate/frame/support/src/lib.rs

```diff
@@ -1019,1221 +916,840 @@ pub mod pallet_prelude {
-/// 		..
-/// 	}
-/// 	```
-/// * use the newest nightly possible.
+/// This approach is beneficial when you use the `include_str` macro at the beginning of
+/// the file and want that documentation to extend to the runtime metadata, without
+/// reiterating the documentation on the pallet module itself.
 pub use frame_support_procedural::pallet;
 
-/// Contains macro stubs for all of the pallet:: macros
-pub mod pallet_macros {
-	pub use frame_support_procedural::{
-		composite_enum, config, disable_frame_system_supertrait_check, error, event,
-		extra_constants, feeless_if, generate_deposit, generate_store, getter, hooks,
-		import_section, inherent, no_default, no_default_bounds, origin, pallet_section,
-		storage_prefix, storage_version, type_value, unbounded, validate_unsigned, weight,
-		whitelist_storage,
-	};
+/// Contains macro stubs for all of the `pallet::` macros
+pub mod pallet_macros {
+
+	/// The `#[pallet::origin]` attribute allows you to define some origin for the pallet.
+	///
+	/// Item must be either a type alias, an enum, or a struct. It needs to be public.
+	///
+	/// ## Example
+	///
+	/// ```
+	/// #[frame_support::pallet]
+	/// mod pallet {
+	/// 	use frame_support::pallet_prelude::*;
+	///
+	/// 	#[pallet::pallet]
+	/// 	pub struct Pallet<T>(_);
+	///
+	/// 	#[derive(PartialEq, Eq, Clone, MaxEncodedLen, Encode, Decode, TypeInfo, RuntimeDebug)]
+	/// 	#[pallet::origin]
+	/// 	pub enum Origin {
+	/// 		/// SomeOrigin doc.
+	/// 		SomeOrigin,
+	/// 		/// AnotherOrigin doc.
+	/// 		AnotherOrigin,
+	/// 	}
+	///
+	/// 	#[pallet::config]
+	/// 	pub trait Config: frame_system::Config {}
+	/// }
+	/// ```
+	///
+	/// **WARNING**: modifying origin changes the outer runtime origin. This outer runtime
+	/// origin can be stored on-chain (e.g. in
+	/// [`pallet_scheduler`](../../pallet_scheduler/index.html), thus any change must be
+	/// done with care as it might require some migration.
+	///
+	/// NOTE: for instantiable pallets, the origin must be generic over `T` and `I`.
+	///
+	/// Read more about origins at the [Origin Reference
+	/// Docs](../../polkadot_sdk_docs/reference_docs/frame_origin/index.html).
+	pub use frame_support_procedural::origin;
+
+	/// The `#[pallet::inherent]` attribute allows the pallet to provide
+	/// [inherents](https://docs.substrate.io/fundamentals/transaction-types/#inherent-transactions).
+	/// An inherent is some piece of data that is inserted by a block authoring node at block
+	/// creation time and can either be accepted or rejected by validators based on whether the
+	/// data falls within an acceptable range.
+	///
+	/// The most common inherent is the `timestamp` that is inserted into every block. Since
+	/// there is no way to validate timestamps, validators simply check that the timestamp
+	/// reported by the block authoring node falls within an acceptable range.
+	///
+	/// Example usage:
+	///
+	/// ```
+	/// #[frame_support::pallet]
+	/// mod pallet {
```

Comment:
btw `"rust-analyzer.semanticHighlighting.doc.comment.inject.enable": true,` is a life changer for writing docs.